### ISSUE TITLE
+[NSLayoutConstraint constraintWithItem:attribute:relatedBy:toItem:attribute:multiplier:constant:] crash

### ISSUE THREAD

**Issue by Raztor0** [Appboy/appboy-ios-sdk#106]
* [x] I have checked [Appboy's documentation](https://www.appboy.com/documentation/iOS/) and am still experiencing the issue.

# Report

## Describe your environment.

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | ios
 Platform Version        | 10.2
 SDK Version             | 2.24.1
 Integration Method      | cocoapods
 Xcode Version           | 8.2
 Repro rate              | 500 crashes in the last 90 days
 
## What did you do?

Seeing this stack trace via Crashlytics:
```
Fatal Exception: NSInternalInconsistencyException
Multiplier is not finite! That's illegal. multiplier:nan
0  CoreFoundation                 0x18af291b8 __exceptionPreprocess
1  libobjc.A.dylib                0x18996055c objc_exception_throw
2  CoreFoundation                 0x18af2908c +[NSException raise:format:]
3  Foundation                     0x18b9e102c -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:]
4  Foundation                     0x18b98a5d8 -[NSLayoutConstraint _setMultiplier:]
5  Foundation                     0x18b98a780 +[NSLayoutConstraint constraintWithItem:attribute:relatedBy:toItem:attribute:multiplier:constant:]
6  RazApp                       0x100968b30 -[ABKInAppMessageModalViewController viewWillAppear:] (ABKInAppMessageModalViewController.m:124)
7  UIKit                          0x190df6754 -[UIViewController _setViewAppearState:isAnimating:]
8  UIKit                          0x190e93010 __52-[UIViewController _setViewAppearState:isAnimating:]_block_invoke
9  CoreFoundation                 0x18af1a6c8 -[__NSSingleObjectArrayI enumerateObjectsWithOptions:usingBlock:]
10 UIKit                          0x190df69dc -[UIViewController _setViewAppearState:isAnimating:]
11 UIKit                          0x190df64cc -[UIViewController __viewWillAppear:]
12 UIKit                          0x190e5aa90 -[UIViewController viewWillMoveToWindow:]
13 UIKit                          0x190de67e0 -[UIView(Hierarchy) _willMoveToWindow:withAncestorView:]
14 UIKit                          0x190de5ab8 -[UIView(Internal) _addSubview:positioned:relativeTo:]
15 UIKit                          0x190de5904 -[UIView(Hierarchy) addSubview:]
16 UIKit                          0x190de4eec -[UIWindow addRootViewControllerViewIfPossible]
17 UIKit                          0x190de2138 -[UIWindow _setHidden:forced:]
18 RazApp                       0x10092f814 __72-[ABKInAppMessageWindowController displayInAppMessageViewWithAnimation:]_block_invoke (ABKInAppMessageWindowController.m:229)
19 libdispatch.dylib              0x189db21fc _dispatch_call_block_and_release
20 libdispatch.dylib              0x189db21bc _dispatch_client_callout
21 libdispatch.dylib              0x189db6d68 _dispatch_main_queue_callback_4CF
22 CoreFoundation                 0x18aed6810 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__
23 CoreFoundation                 0x18aed43fc __CFRunLoopRun
24 CoreFoundation                 0x18ae022b8 CFRunLoopRunSpecific
25 GraphicsServices               0x18c8b6198 GSEventRunModal
26 UIKit                          0x190e497fc -[UIApplication _run]
27 UIKit                          0x190e44534 UIApplicationMain
28 RazApp                       0x10003b448 main (main.m:13)
29 libdispatch.dylib              0x189de55b8 (Missing)
```

## What did you expect to happen?

In-app message with imageStyle ABKInAppMessageGraphic is shown.

## What happened instead?

The above crash.

## Steps to reproduce

Unfortunately the only information I have is the stack trace I'm seeing.

Although, upon disassembling [ABKInAppMessageModalViewController viewWillAppear:], it seems that this crash is only possible in the case that the imageStyle is set to ABKInAppMessageGraphic, since in the other case, the NSLayoutConstraint is initialized with a constant value (which can never be infinity) instead of something which is computed from the result of a division.

## Code Snippet

Cannot repro this myself

## Are you doing any feature customizations that may relate to the issue? Can you share the code snippet?

No customizations

## Project that demonstrates the issue

Don't have one

**Comment by Wenzhi** [comment]
Hey @Raztor0, 

Sorry that this issue has caused so many crashes! The issue looks very similar to the one in #102, for which we released a patch in SDK version 2.24.4 and solved it from the root in 2.26.0. Could you try out our latest SDK and see if you still see the same error?

Thanks for reporting the issue with so much detail to us. Please let us know if the latest SDK stops the crashes.

Thanks,
Wenzhi

**Comment by Raztor0** [comment]
Thanks @Wenzhi, we'll upgrade the SDK and let you know if we continue seeing this issue in the future.
